src/core/parser.js

```diff
@@ -49,662 +50,705 @@ function computeAdler32(bytes) {
   return ((b % 65521) << 16) | (a % 65521);
 }
 
-var Parser = (function ParserClosure() {
-  function Parser(lexer, allowStreams, xref, recoveryMode) {
+class Parser {
+  constructor(lexer, allowStreams, xref, recoveryMode = false) {
```

Comment:
Another possible improvement here, in a separate commit or PR, would be to change the constructor to take an object instead to get rid of the often "ugly" call-sites (and document them for free):
```javascript
constructor({ lexer, xref, allowStreams = false, recoveryMode = false, }) {
```

Reply:
I agree, I'll see what I can do.

Reply:
Since this patch is already quite big, I'd like to leave this for a possible follow-up.

Reply:
> Since this patch is already quite big, I'd like to leave this for a possible follow-up.

That's completely fine; just as long as it eventually happens, given how unwieldy the call-sites currently look :-)